pkg/config/options.go

```diff
@@ -34,11 +41,58 @@ func (o *Options) Validate() error {
 	return nil
 }
 
+func (o *Options) Complete() error {
+	if o.onlyProcessChanges {
+		_, err := exec.LookPath("git")
+		if err != nil {
+			return fmt.Errorf("couldn't find git command in the system: %w", err)
+		}
+
+		mainBranch, err := getUpstreamBranch()
+		if err != nil {
+			return fmt.Errorf("couldn't get upstream branch: %w", err)
+		}
+
+		cmd := exec.Command("git", "diff", "--name-status", mainBranch)
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("couldn't get the changed files comparing with %s branch: %w", mainBranch, err)
+		}
+
+		trimmedLastLine := strings.TrimSuffix(string(out), "\n")
+		changedFilesWithStatus := strings.Split(trimmedLastLine, "\n")
+
+		// Fill the modified files set based on the git output. If we detect anything else than modified files,
+		// turn onlyProcessChanges to false, in order to process all the files instead of a subset. For now we
+		// care of processing on modified files when there are ONLY modified files.
+		// Example output of `git diff --name-status master`
+		// M       path/of/modified/file.yaml
+		// D       path/of/deleted/file.yaml
+		// R100    path/of/renamed/old-name.yaml   path/of/renamed/new-name.yaml
+		o.modifiedFiles = sets.NewString()
+		for _, f := range changedFilesWithStatus {
+			statusAndFile := strings.Fields(f)
+			if len(statusAndFile) > 1 && statusAndFile[0] == "M" {
+				o.modifiedFiles.Insert(statusAndFile[1])
+				continue
+			}
+			o.onlyProcessChanges = false
+		}
+	}
+	return nil
+}
+
+func (o *Options) IsOnlyProcessChanges() bool {
```

Comment:
```suggestion
func (o *Options) ProcessAll() bool {
```

Then your callsite becomes

```go
if opt.ProcessAll()
```